server/game/cards/locations/04/harrenhalffh.js

```diff
@@ -1,38 +1,36 @@
 const DrawCard = require('../../../drawcard.js');
 
 class Harrenhal extends DrawCard {
-    setupCardAbilities() {
+    setupCardAbilities(ability) {
         this.action({
             title: 'Kneel Harrenhal to put a character in play',
-            method: 'kneel',
-            phase: 'challenge'
+            cost: ability.costs.kneelSelf(),
+            phase: 'challenge',
+            handler: context => {
```

Comment:
The intention was to have `handler` to replace `method`, since it is far more consistent with the way reactions and interrupts are done. It can also open the door to prefab handlers later (e.g general 'draw card', 'gain gold', 'kill selected character', etc).

If we do start converting them back to `method`, keep in mind that the signature is currently different between the two - handler takes a context, while the method version takes player, arg, then context. The context object **is** necessary as it provides information about the costs paid (which can sometimes involve player choices). This difference in signatures was done to ensure backward compatibility while things were updated to use `handler` instead.

If that's not the direction we're going, a concerted effort should be made to have method just take a context instead (aka update every single usage in a single PR).